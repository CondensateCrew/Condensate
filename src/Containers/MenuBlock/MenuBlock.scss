@import "../../css/variables.scss";

.menu-block {
  display: -webkit-flex;
  display: -ms-flex;
  display: flex;
  justify-content: flex-end;
  background: $primary-gradient;
  box-shadow: $external-shadow;
  border-radius: $one-line-border-radius;
  padding: $one-line-padding-space;
  position: absolute;
  right: 0; top: 0;
  img {
    width: $medium-text;
    margin-left: $small-space;
    &:last-child {
      opacity: 0.4;
      margin-left: $medium-space;
      &:hover {
        opacity: 1;
      }
    }
  }
}
