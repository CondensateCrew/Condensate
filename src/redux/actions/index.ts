import {
  Action,
  Category,
  Brainstorm,
  UserLoginReceived,
  RandomWordCollection,
  Insight,
  ActionObject,
  IAddUserAction,
  ILogoutUserAction,
  IAddAllActionsAction,
  IAddNewActionAction,
  IAddAllBrainstormsAction,
  IAddNewBrainstormAction,
  IDeleteBrainstormAction,
  IAddAllCategoriesAction,
  IAddNewCategoryAction,
  IAddRandomWordCollectionsAction,
  IAddQuestionTemplatesAction,
  IAddSecretSauceAction,
  IAddCurrentBrainstormAction,
  IRemoveCurrentBrainstormAction,
  IAddChosenWordAction,
  IRemoveChosenWordAction,
  IAddInsightAction,
  IAddQueryAction,
  IRemoveQueryAction,
  IAddFilterAction,
  IRemoveFilterAction,
  IEndTimeAction,
  IReverseTimeAction,
  IEndInstructionsAction,
  IReverseInstructionsAction
} from 'interfaces';

// user reducer action creators
export const addUser = (user: UserLoginReceived): IAddUserAction => ({
  type: 'ADD_USER',
  user: user
});

export const logOutUser = (): ILogoutUserAction => ({
  type: 'LOG_OUT_USER',
});

// actions reducer action creators
export const addAllActions = (actions: Action[]): IAddAllActionsAction => ({
  type: 'ADD_ALL_ACTIONS',
  actions: actions
});

export const addNewAction = (action: Action): IAddNewActionAction => ({
  type: 'ADD_NEW_ACTION',
  action: action
});

// allBrainstorms reducer action creators
export const addAllBrainstorms = (brainstorms: Brainstorm[]): IAddAllBrainstormsAction => ({
  type: 'ADD_ALL_BRAINSTORMS',
  brainstorms: brainstorms
});

export const addNewBrainstorm = (brainstorm: Brainstorm): IAddNewBrainstormAction => ({
  type: 'ADD_BRAINSTORM',
  brainstorm: brainstorm
});

export const deleteBrainstorm = (id: number): IDeleteBrainstormAction => ({
  type: 'DELETE_BRAINSTORM',
  id: id
});

// categories reducer action creators
export const addAllCategories = (categories: Category[]): IAddAllCategoriesAction => ({
  type: 'ADD_ALL_CATEGORIES',
  categories: categories
});

export const addNewCategory = (category: Category): IAddNewCategoryAction => ({
  type: 'ADD_CATEGORY',
  category: category
});

// randomWordCollections reducer action creators
export const addRandomWordCollections = (collections: RandomWordCollection[]): IAddRandomWordCollectionsAction => ({
  type: 'ADD_ALL_COLLECTIONS',
  collections: collections
});

// questionTemplates reducer action creators
export const addQuestionTemplates = (templates: string[]): IAddQuestionTemplatesAction => ({
  type: 'ADD_ALL_TEMPLATES',
  templates: templates
});

// secretSauce reducer action creators
export const addSecretSauce = (secretSauce: string[]): IAddSecretSauceAction => ({
  type: 'ADD_SECRETE_SAUSE',
  secretSauce: secretSauce
});

// currentBrainstorm reducer action creators
export const addCurrentBrainstorm = (currentBrainstorm: Brainstorm): IAddCurrentBrainstormAction => ({
  type: 'ADD_CURRENT_BRAINSTORM',
  currentBrainstorm: currentBrainstorm
});

export const removeCurrentBrainstorm = (): IRemoveCurrentBrainstormAction => ({
  type: 'REMOVE_CURRENT_BRAINSTORM'
});

// chosenWords reducer action creators
export const addChosenWord = (chosenWord: string): IAddChosenWordAction => ({
  type: 'ADD_WORD',
  chosenWord: chosenWord
});

export const removeChosenWord = (chosenWord: string): IRemoveChosenWordAction => ({
  type: 'REMOVE_WORD',
  chosenWord: chosenWord
});

// insights reducer action creators
export const addInsight = (insight: Insight): IAddInsightAction => ({
  type: 'ADD_INSIGHT',
  insight: insight
});

// query reducer action creators
export const addQuery = (query: string): IAddQueryAction => ({
  type: 'ADD_QUERY',
  query: query
});

export const removeQuery = (): IRemoveQueryAction => ({
  type: 'REMOVE_QUERY'
});

// filter reducer action creators
export const addFilter = (filter: string): IAddFilterAction => ({
  type: 'ADD_FILTER',
  filter: filter
});

export const removeFilter = (): IRemoveFilterAction => ({
  type: 'REMOVE_FILTER'
});

// timeEnded reducer action creators
export const endTime = (): IEndTimeAction => ({
  type: 'END_TIME'
});

export const reverseTime = (): IReverseTimeAction => ({
  type: 'REVERSE_TIME'
});

// instructionsEnded reducer action creators
export const endInstructions = (): IEndInstructionsAction => ({
  type: 'END_INSTRUCTIONS'
});

export const reverseInstructions = (): IReverseInstructionsAction => ({
  type: 'REVERSE_INSTRUCTIONS'
});
