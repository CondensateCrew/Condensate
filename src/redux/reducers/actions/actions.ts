import { ActionObject } from 'interfaces';

type state = string[] | [ ];

const actions = (state: state = [ ], action: ActionObject) => {
  switch(action.type) {
    case 'ADD_ALL_ACTIONS':
      return action.actions;
    case 'ADD_NEW_ACTION':
      return [...state, action.action];
    case 'CLEAN_STORE':
      return [ ];
    default:
      return state;
  }
}

export default actions;
