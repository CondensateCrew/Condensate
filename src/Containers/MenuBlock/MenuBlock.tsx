import './MenuBlock.scss';
import React from 'react';
import { useDispatch } from "react-redux";
import { toggleBrainstorm, deleteBrainstorm} from 'redux/actions'
import ideaIcon from 'assets/idea.svg';
import closeIcon from 'assets/close.svg';
import deleteIcon from 'assets/delete.svg';

interface Props {
  id: number,
  setIsClicked: (isClicked: boolean) => void;
};

const MenuBlock: React.FC<Props> = ({setIsClicked, id}) => {
  const closeMenu = (): void => setIsClicked(false);
  const dispatch = useDispatch();
  const toggleCard = () => dispatch(toggleBrainstorm(id));
  const deleteCard = () => dispatch(deleteBrainstorm(id));
  return (
    <div className="menu-block">
      <img src={ideaIcon} alt="idea" onClick={toggleCard}/>
      <img src={deleteIcon} alt="delete" onClick={deleteCard}/>
      <img src={closeIcon} alt="close" onClick={closeMenu}/>
    </div>
  )
}

export default MenuBlock;
