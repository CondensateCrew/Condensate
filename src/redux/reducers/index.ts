import { combineReducers } from 'redux';
import user from '../reducers/user/user';
import allBrainstorms from '../reducers/allBrainstorms/allBrainstorms';
import categories from '../reducers/categories/categories';
import actions from '../reducers/actions/actions';
import randomWordCollections from '../reducers/randomWordCollections/randomWordCollections';
import questionTemplates from '../reducers/questionTemplates/questionTemplates';
import secretSauce from '../reducers/secretSauce/secretSauce';
import currentBrainstorm from '../reducers/currentBrainstorm/currentBrainstorm';
import chosenWords from '../reducers/chosenWords/chosenWords';
import insights from '../reducers/insights/insights';
import query from '../reducers/query/query';
import filter from '../reducers/filter/filter';
import timeEnded from '../reducers/timeEnded/timeEnded';
import instructionsEnded from '../reducers/instructionsEnded/instructionsEnded';

const rootReducer = combineReducers({
  user,
  allBrainstorms,
  categories,
  actions,
  randomWordCollections,
  questionTemplates,
  secretSauce,
  currentBrainstorm,
  chosenWords,
  insights,
  query,
  filter,
  timeEnded,
  instructionsEnded
});

export default rootReducer;
