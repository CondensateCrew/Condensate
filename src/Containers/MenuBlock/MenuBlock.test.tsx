import React from 'react';
import MenuBlock from './MenuBlock';
import { shallow } from 'enzyme';
import { toggleBrainstorm, deleteBrainstorm} from 'redux/actions';

jest.mock('redux/actions')

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch
}));

describe('MenuBlock component', () => {
  let wrapper: any;
  const setIsClicked = jest.fn();

  const mockProps = {
    id: 1,
    setIsClicked
  };

  beforeEach(() => {
    wrapper = shallow(<MenuBlock {...mockProps}/>);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call setIsClicked prop when close btn is clicked', () => {
    wrapper.find('img[alt="close"]').simulate('click');
    expect(setIsClicked).toHaveBeenCalledWith(false);
  });

  it('should call toggleBrainstorm prop when idea btn is clicked', () => {
    wrapper.find('img[alt="idea"]').simulate('click');
    expect(toggleBrainstorm).toHaveBeenCalledWith(1);
  });

  it('should call deleteBrainstorm prop when delete btn is clicked', () => {
    wrapper.find('img[alt="delete"]').simulate('click');
    expect(deleteBrainstorm).toHaveBeenCalledWith(1);
  });
});
