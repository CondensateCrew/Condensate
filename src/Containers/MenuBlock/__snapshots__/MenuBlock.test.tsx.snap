// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuBlock component should match the snapshot 1`] = `
<div
  className="menu-block"
>
  <img
    alt="idea"
    onClick={[Function]}
    src="idea.svg"
  />
  <img
    alt="delete"
    onClick={[Function]}
    src="delete.svg"
  />
  <img
    alt="close"
    onClick={[Function]}
    src="close.svg"
  />
</div>
`;
